iothub/service/src/AmqpServiceClient.cs

```diff
@@ -106,6 +116,11 @@ public async override Task SendAsync(string deviceId, Message message, TimeSpan?
             {
                 SendingAmqpLink sendingLink = await GetSendingLinkAsync().ConfigureAwait(false);
 
+                if (Logging.IsEnabled)
+                {
+                    Logging.Info(this, $"Retrieved SendingAmqpLink [{sendingLink.Name}]", nameof(SendAsync));
```

Comment:
What if sendingLink is null?

Reply:
It cannot be, this is a result of `_faultTolerantSendingLink.GetOrCreateAsync()` which will give us a link, or throw an exception (unless there is a bug in the amqp lib). But this is a good point, I'll add a null check here and also I'll be adding more logs in our service client that will log the state for these link creation as well.